pkg/tasks/tasks.go

```diff
@@ -21,43 +21,53 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// TaskRunner executes groups of tasks, the groups sequentially, each group concurrently
```

Comment:
nit: semantically, the struct itself does not execute the group of tasks -I would guess the `RunAll` function does that?

Reply:
Fair point, add more comments and shuffled stuff around.